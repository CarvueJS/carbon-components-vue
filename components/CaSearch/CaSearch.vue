<template>
  <div data-search class="bx--search bx--search--sm" role="search">
    <svg class="bx--search-magnifier" width="16" height="16" viewBox="0 0 16 16" fill-rule="evenodd">
      <path d="M6 2c2.2 0 4 1.8 4 4s-1.8 4-4 4-4-1.8-4-4 1.8-4 4-4zm0-2C2.7 0 0 2.7 0 6s2.7 6 6 6 6-2.7 6-6-2.7-6-6-6zM16 13.8L13.8 16l-3.6-3.6 2.2-2.2z"></path>
      <path d="M16 13.8L13.8 16l-3.6-3.6 2.2-2.2z"></path>
    </svg>
    <label id="search-input-label-1" class="bx--label" for="search__input-2">Search</label>
    <input class="bx--search-input" type="text" id="search__input-2" role="search" placeholder="Search" aria-labelledby="search-input-label-1">
    <svg class="bx--search-close bx--search-close--hidden" width="16" height="16" viewBox="0 0 16 16" fill-rule="evenodd">
      <path d="M8 0C3.6 0 0 3.6 0 8s3.6 8 8 8 8-3.6 8-8-3.6-8-8-8zm3.5 10.1l-1.4 1.4L8 9.4l-2.1 2.1-1.4-1.4L6.6 8 4.5 5.9l1.4-1.4L8 6.6l2.1-2.1 1.4 1.4L9.4 8l2.1 2.1z"></path>
    </svg>
  </div>
</template>

<script>
  export default {
    name: 'ca-search'
  }
</script>
