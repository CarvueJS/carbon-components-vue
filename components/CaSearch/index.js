import CaSearch from './CaSearch.vue';

CaSearch.install = Vue => {
  Vue.component(CaSearch.name, CaSearch);
}

export default CaSearch
