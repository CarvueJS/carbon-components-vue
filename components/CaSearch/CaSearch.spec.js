import { shallow } from 'vue-test-utils';
import CaSearch from './CaSearch.vue';

describe('CaSearch', () => {
  test('is a Vue instance', () => {
    const wrapper = shallow(CaSearch);
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test('Mathes snapshot', () => {
    const wrapper = shallow(CaSearch);
    expect(wrapper.html()).toMatchSnapshot()
  });
})
