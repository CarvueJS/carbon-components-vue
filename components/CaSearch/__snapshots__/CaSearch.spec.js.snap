// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaSearch Mathes snapshot 1`] = `
<div data-search="" role="search" class="bx--search bx--search--sm">
  <svg width="16" height="16" viewBox="0 0 16 16" fill-rule="evenodd" class="bx--search-magnifier">
    <path d="M6 2c2.2 0 4 1.8 4 4s-1.8 4-4 4-4-1.8-4-4 1.8-4 4-4zm0-2C2.7 0 0 2.7 0 6s2.7 6 6 6 6-2.7 6-6-2.7-6-6-6zM16 13.8L13.8 16l-3.6-3.6 2.2-2.2z"></path>
    <path d="M16 13.8L13.8 16l-3.6-3.6 2.2-2.2z"></path>
  </svg>
  <label id="search-input-label-1" for="search__input-2" class="bx--label">Search</label>
  <input type="text" id="search__input-2" role="search" placeholder="Search" aria-labelledby="search-input-label-1" class="bx--search-input">
  <svg width="16" height="16" viewBox="0 0 16 16" fill-rule="evenodd" class="bx--search-close bx--search-close--hidden">
    <path d="M8 0C3.6 0 0 3.6 0 8s3.6 8 8 8 8-3.6 8-8-3.6-8-8-8zm3.5 10.1l-1.4 1.4L8 9.4l-2.1 2.1-1.4-1.4L6.6 8 4.5 5.9l1.4-1.4L8 6.6l2.1-2.1 1.4 1.4L9.4 8l2.1 2.1z"></path>
  </svg>
</div>
`;
